examples/flows/standard/flow-with-assistant/add_message_and_run.py

```diff
@@ -0,0 +1,155 @@
+import asyncio
+import json
+
+from openai import AsyncOpenAI
+from openai.types.beta.threads import MessageContentImageFile, MessageContentText
+
+from promptflow import tool
+from promptflow.connections import OpenAIConnection
+from promptflow.contracts.multimedia import Image
+from promptflow.exceptions import SystemErrorException
+from promptflow.executor._assistant_tool_invoker import AssistantToolInvoker
+
+URL_PREFIX = "https://platform.openai.com/files/"
+
+
+@tool
+async def add_message_and_run(
+    conn: OpenAIConnection,
+    assistant_id: str,
+    thread_id: str,
+    message: list,
+    assistant_definition: dict,
+    download_images: bool
+):
+    content = extract_text_from_message(message)
+    file_ids = await extract_file_ids_from_message(message, conn)
+    cli = AsyncOpenAI(api_key=conn.api_key, organization=conn.organization)
+    await cli.beta.threads.messages.create(
+        thread_id=thread_id,
+        role="user",
+        content=content,
+        file_ids=file_ids
+    )
+    invoker = AssistantToolInvoker()
+    invoker.load_tools(assistant_definition["tools"])
+    run = await cli.beta.threads.runs.create(
```

Comment:
A P2 ask. 

When a user clone this sample, there is no way to have a valid assistant id as input. 

We may think about to create an assistant with assistant definition if assistant id is empty.

Reply:
done, integrated into assisant tool if assistant id is missing, auto create one